src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -2,42 +2,59 @@ using Oceananigans.Advection:
     _advective_momentum_flux_Uu,
     _advective_momentum_flux_Uv,
     _advective_momentum_flux_Vu,
-    _advective_momentum_flux_Vv
+    _advective_momentum_flux_Vv,
+    advective_tracer_flux_x, 
+    advective_tracer_flux_y,
+    vertical_vorticity_U,
+    vertical_vorticity_V,
+    bernoulli_head_U,
+    bernoulli_head_V
 
 using Oceananigans.Grids: AbstractGrid
 using Oceananigans.Operators: Ax_qᶠᶜᶜ, Ay_qᶜᶠᶜ
 
+
 #####
 ##### Momentum flux operators
 #####
 
 @inline momentum_flux_huu(i, j, k, grid, advection, solution) =
-    @inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
+    @inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, solution[1], solution[1]) / solution.h[i, j, k]
 
 @inline momentum_flux_hvu(i, j, k, grid, advection, solution) =
-    @inbounds _advective_momentum_flux_Vu(i, j, k, grid, advection, solution.vh, solution.uh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
+    @inbounds _advective_momentum_flux_Vu(i, j, k, grid, advection, solution[2], solution[1]) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
 @inline momentum_flux_huv(i, j, k, grid, advection, solution) =
-    @inbounds _advective_momentum_flux_Uv(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
+    @inbounds _advective_momentum_flux_Uv(i, j, k, grid, advection, solution[1], solution[2]) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
 @inline momentum_flux_hvv(i, j, k, grid, advection, solution) =
-    @inbounds _advective_momentum_flux_Vv(i, j, k, grid, advection, solution.vh, solution.vh) / solution.h[i, j, k]
+    @inbounds _advective_momentum_flux_Vv(i, j, k, grid, advection, solution[2], solution[2]) / solution.h[i, j, k]
 
 #####
 ##### Momentum flux divergence operators
 #####
 
-@inline div_hUu(i, j, k, grid, advection, solution) =
-    1 / Vᶠᶜᶜ(i, j, k, grid) * (δxᶠᵃᵃ(i, j, k, grid, momentum_flux_huu, advection, solution) +
+@inline div_hUu(i, j, k, grid, advection, solution, formulation) =
+    1 / Vᶜᶜᶜ(i, j, k, grid) * (δxᶠᵃᵃ(i, j, k, grid, momentum_flux_huu, advection, solution) +
                                δyᵃᶜᵃ(i, j, k, grid, momentum_flux_hvu, advection, solution))
 
-@inline div_hUv(i, j, k, grid, advection, solution) =
-    1 / Vᶜᶠᶜ(i, j, k, grid) * (δxᶜᵃᵃ(i, j, k, grid, momentum_flux_huv, advection, solution) +
+@inline div_hUv(i, j, k, grid, advection, solution, formulation) =
+    1 / Vᶜᶜᶜ(i, j, k, grid) * (δxᶜᵃᵃ(i, j, k, grid, momentum_flux_huv, advection, solution) +
```

Comment:
Do we want to change this `V` to `Az` also?

Reply:
I thiknk so.